Dockerfile

```diff
@@ -8,10 +8,12 @@ ARG GID=1000
 
 RUN bash -c "set -o pipefail && apt-get update \
   && apt-get install -y --no-install-recommends build-essential curl git libpq-dev \
-  && curl -sSL https://deb.nodesource.com/setup_20.x | bash - \
-  && curl -sSL https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
-  && echo 'deb https://dl.yarnpkg.com/debian/ stable main' | tee /etc/apt/sources.list.d/yarn.list \
-  && apt-get update && apt-get install -y --no-install-recommends nodejs yarn \
+  && apt-get install -y ca-certificates curl gnupg \
+  && mkdir -p /etc/apt/keyrings \
+  && curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg \
```

Comment:
I think we can optimize this to not need `gpg` which means we can remove `ca-certificates` and `gnupg` as dependencies such as deleting the other `&& apt-get install -y ca-certificates curl gnupg \` line entirely.

For example, this works in the Dockerfile:

```sh
curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key -o /etc/apt/keyrings/nodesource.asc \
  &&  echo "deb [signed-by=/etc/apt/keyrings/nodesource.asc] https://deb.nodesource.com/node_20.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list \
  && apt-get update && apt-get install -y --no-install-recommends nodejs \
  && corepack enable \
```

Which produces this in the container:

```sh
$ node --version
v20.6.1

$ yarn --version
1.22.19

$ apt-get update
Hit:1 http://deb.debian.org/debian bookworm InRelease
Hit:2 http://deb.debian.org/debian bookworm-updates InRelease
Hit:3 http://deb.debian.org/debian-security bookworm-security InRelease
Hit:4 https://deb.nodesource.com/node_20.x nodistro InRelease
Reading package lists... Done
```

Reply:
Nice! Yep, that worked like a charm.

Reply:
That works because with a `.asc` file extension it'll use the plain text version of the key where as `.gpg` will use the binary version of the key. This was top of mind because I recently updated my Ansible Docker role to do the same thing.